src/WebJobs.Script/Host/IFunctionMetadataManager.cs

```diff
@@ -2,14 +2,14 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System.Collections.Immutable;
-using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Azure.WebJobs.Script.Abstractions.Description;
 
 namespace Microsoft.Azure.WebJobs.Script
 {
     public interface IFunctionMetadataManager
     {
         ImmutableDictionary<string, ImmutableArray<string>> Errors { get; }
 
-        ImmutableArray<FunctionMetadata> Functions { get; }
+        ImmutableArray<FunctionMetadata> GetFunctionMetadata(bool forceRefresh = false, bool includeBlocked = false);
```

Comment:
If this also took predicate (Func over the metadata items), you'd be able to simplify the calls to the utility methods.

Reply:
With the other suggested, I think this may not be necessary as I ended up being able to remove Utility Functions altogether. And callers can call `.Where(Func<T,bool> predicate)` if they need to filter.

Not having it there, avoids having to re-create another ImmutableArray after filtering to fulfil the return type contract.

Reply:
Here also we're making breaking changes to public types, because we're pretty confident we're the only one using them